### PR TITLE
Adds a CORS endpoint to the server to use for demos.

### PR DESCRIPTION
**Pull request by rachelrosengoogle**
API requests from the demo plugin (or any host from a different origin than the server) should hit the /plugin/check endpoint which accepts CORS. Direct API requests from a host on the same server should hit the /check endpoint.

### CODE REVIEW

File: src/serving.ts [hunk 1]
```diff
@@ -68,21 +70,7 @@ export class Server {
...
-    //   // Selects the log to write to
-    //   let logWrite = loggingClient.log(LOG_NAME).write;
-    //   this.log = { write : (s:string) :void => {
-    //     // We wrap this just to catch the error so that node doesn't crash
-    //     // should stackdriver fail a log statement.
-    //     // Once we confirm that stackdriver log functions never fail their promise,
-    //     // then this can be removed.
-    //     logWrite(s).catch((e:Error) => { console.error(e); });
-    //   } };
+      // TODO(ldixon): do the cloud logging thing...
```
Comment (iislucas) [inline src/serving.ts]: You can delete this comment too now. Cloud logging changed and fixed the bad bugs I that were problematic here before.

**Comment by iislucas** [comment]
Please add a description of what this intended to do. e.g. what you want to happen to what URLs.

File: src/serving.ts [hunk 2]
```diff
@@ -129,55 +106,22 @@ export class Server {
...
-        if (context.article_and_parent_comment &&
-            requestData.parentComment) {
-          context.article_and_parent_comment.parent_comment = {
-            text: requestData.parentComment
-          };
-        }
-      }
+    // Define an endpoint with CORS enabled to be used by the plugin.
+    this.app.options('/demo/check', cors()) // enable CORS for pre-flight requests.
+    this.app.post('/demo/check', cors(), (req, res) => {
```
Comment (iislucas) [inline src/serving.ts]: If you specify cores in the options for a URL prefix, you should not need it in the post option also.

File: src/serving.ts (hunk 1 above)
Comment (rachelrosengoogle) [reply src/serving.ts]: Done

File: src/serving.ts (hunk 2 above)
Comment (rachelrosengoogle) [reply src/serving.ts]: I just tried it, looks like it is required for both preflight and actual POST request.

Comment (iislucas) [reply src/serving.ts]: Oh! OK, I stand corrected! Good to know!

**Review by iislucas — APPROVED** [review]

**Pull request merged**